index.js

```diff
@@ -33,8 +33,9 @@ function errorHandler(app) {
     if (app.get('env') == 'dev' && !err.statusCode) {
       throw err;
     }
-
-    res.status(err.statusCode || 500).send({message: err.statusCode ? err.message : 'Internal Server Error'});
+    err.message = err.statusCode ? err.message : 'Internal Server Error'
+    
+    res.status(err.statusCode || 500).send({message: err.message, userMessage: err.userMessage || ''});
```

Comment:
why dont we return the userMessage in the `message` field? I think it's kind of weird we return 2 messages

Reply:
The following snippet is an example of how userMessage is helping the client:
```

Reply:
There's no snippet :)

Reply:
The following snippet is an example of how userMessage is helping the client:
```js
} catch(err){
      this.props.update({
        notification: {
          message: err.userMessage || 'unknownError',
          timeout: 4000
        }
      });

      console.log(err);
    }
```
Any error could happen in the `try block`, it could be a proper 400 from the API, in which case `err.message` will be `You cannot add more than 12 items to cart` but it could also another error where `err.message` is `Unexpected end of JSON input`. You would agree that the former should be shown to the user but not the latter :). Now, how do we make that decision if both can come in `err.message`? This is the problem that `err.userMessage` solves.

By the way, you can read more about the `userMessage` approach here http://blog.restcase.com/rest-api-error-codes-101/ :)

Reply:
Actually I don't like how we are restricting our error response to a certain format. One application might need a `userMessage`, another a `stacktrace` or `errorCode` or else. I think we should allow the user of this library to send whatever he sees appropriate for his application. I would do something like this:
```
  if (err.data) {
    res.status(err.statusCode || 500).send(err.data)
    return
  }

  res.status(err.statusCode || 500).send({message: err.statusCode ? err.message : 'Internal Server Error'})
```
then to send a user message I would do:
```
let error = httpError(400);
error.data = {userMessage: 'some message to the user'};
throw error;
```

Reply:
@alahawash I understand your point. My only concern is how would we make that backward compatible? If I change an existing API to use `err.data` and then deploy it, it will break clients that are still expecting `err.message`. One way to avoid this issue would be to keep sending `err.message` along with `err.data`. What do you guys think? cc @AyhamAlzoubi

Reply:
in any case, we need to unify the way we send the errors to the client. for @alahawash suggestion is not bad, but we should send the error as a message, not as data.

I saw a lot of ppl use userMessage and developerMessage. For us, we can either keep [developerMessage, message] or [message, userMessage] they are the same, and I think this solution is simpler for the clients they can unify the error handling for our api calls and the userMessage should be translated from the api.

@joejean why don't we return `userMessage: err.userMessage || 'translated default message'`?

Reply:
@joejean my suggestion is backward compatible, it still uses `err.message` as the default response, but would send `err.data` whenever it is set on the error object.